chainforge/react-server/src/LLMListComponent.js

```diff
@@ -355,34 +357,46 @@ export const LLMListContainer = forwardRef(function LLMListContainer(
     [bgColor],
   );
 
+  const parents = [];
+  const menuItems = [];
+  for (const item of AvailableLLMs) {
+    if (!item.parent) {
+      menuItems.push({
+        key: item.model,
+        title: `${item.emoji} ${item.name}`,
+        onClick: () => handleSelectModel(item.base_model),
+      });
+    } else if (!parents.includes(item.parent)) {
+      parents.push(item.parent);
+      menuItems.push({
+        key: item.model,
+        title: item.parent,
+        onClick: () => handleSelectModel(item.base_model),
+        items: AvailableLLMs.filter(
+          (subItem) => subItem.parent === item.parent,
+        ).map((subItem) => ({
+          key: subItem.model,
+          title: `${subItem.emoji} ${subItem.name}`,
+          onClick: () => handleSelectModel(subItem.base_model),
+        })),
+      });
+    }
+
+    // return undefined;
+  }
+
   return (
     <div className="llm-list-container nowheel" style={_bgStyle}>
       <div className="llm-list-backdrop" style={_bgStyle}>
         {description || "Models to query:"}
         <div className="add-llm-model-btn nodrag">
-          <Menu
-            transitionProps={{ transition: "pop-top-left" }}
-            position="bottom-start"
-            width={220}
-            withinPortal={true}
+          <button
+            style={_bgStyle}
+            onClick={showContextMenu(menuItems)}
+            onContextMenu={showContextMenu(menuItems)}
```

Comment:
For some reason (a bug with Mantine ContextMenu), the context menu was not closing upon click-off, which is highly annoying to the user. It was also mispositioned at the location of the user click rather than nicely placed below the button. I've fixed these issues in an update but it took about 2 hours to do, as the solution was rather hacky.